### ISSUE TITLE
Inconsistent `mode="auto"` behavior in `ModelCheckpoint` vs `EarlyStopping` and `ReduceLROnPlateau`

### ISSUE THREAD

**Issue by adar21a** [keras-team/keras#21118]
When using the `mode="auto"` (default) setting in the ModelCheckpoint callback, the behavior differs from that of EarlyStopping and ReduceLROnPlateau.

**Current Behavior:**
- **ModelCheckpoint:**
"In "auto" mode, the mode is set to "max" if the quantities monitored are "acc" or start with "fmeasure" and are set to "min" for the rest of the quantities."
- **EarlyStopping and ReduceLROnPlateau:**
"In "auto" mode, the direction is automatically inferred from the name of the monitored quantity."

**Problem:**
This inconsistency can lead to unexpected behavior when switching between callbacks or using them together. For example, if a user monitors `AUC` metric, ModelCheckpoint mode will default to "min" while EarlyStopping correctly infers the direction to "max".

**Proposed Change:**
To ensure consistent behavior across callbacks, I propose replacing the `mode="auto"` logic in ModelCheckpoint with the generalized logic used in EarlyStopping.

**Benefits:**
- Aligns behavior across commonly used callbacks.
- Reduces unexpected bugs due to diverging default behaviors.
- Promotes maintainability by sharing consistent logic.

**Comment by adar21a** [comment]
```python

import keras
import numpy as np


# Define the callbacks with the same monitored metric
checkpoint = keras.callbacks.ModelCheckpoint(
    "MODEL_PATH.keras",
    monitor="auc",
    mode="auto"
    )

earlystop = keras.callbacks.EarlyStopping(
    monitor="auc",
    mode="auto"
    )

# Define the Keras model to add callbacks to
def get_model():
    model = keras.Sequential()
    model.add(keras.layers.Dense(1))
    model.compile(
        optimizer="adam",
        loss="binary_crossentropy",
        metrics=["auc"],
    )
    return model

# Define dummy data
x_train = np.ones([5, 10])
y_train = np.zeros(5)

model = get_model()
model.fit(
    x_train,
    y_train,
    epochs=1,
    callbacks=[checkpoint, earlystop],
)

# The `monitor_op` of AUC metric should be `greater`
assert earlystop.monitor_op == keras.src.ops.numpy.greater

assert checkpoint.monitor_op == np.greater
```

This example will raise `AssertionError` on `checkpoint.monitor_op == np.greater`, because it equal `np.less`.
